components/post_view/embedded_bindings/select_binding/select_binding.tsx

```diff
@@ -80,7 +84,31 @@ export default class SelectBinding extends React.PureComponent<Props, State> {
             post.id,
         );
 
-        this.props.actions.doAppCall(call);
+        const res = await this.props.actions.doAppCall(call);
+        const callResp = (res as {data: AppCallResponse}).data;
+        const ephemeral = (message: string) => sendEphemeralPost(message, this.props.post.channel_id, this.props.post.root_id);
+        switch (callResp.type) {
+        case AppCallResponseTypes.OK:
+            if (callResp.markdown) {
+                ephemeral(callResp.markdown);
+            }
+            break;
+        case AppCallResponseTypes.ERROR: {
+            const errorMessage = callResp.error || this.props.intl.formatMessage({id: 'apps.error.unknown', defaultMessage: 'Unknown error happenned'});
```

Comment:
curious, in what circumstances can we have `AppCallResponseTypes.ERROR` but `callResp.error` not set?

Reply:
In theory, only at the doAppCall level, in case the app returns a malformed response (error type with no error text).
I still make the check everywhere else to be extra cautious. Potentially, we can assume that error will be always set at this point, and move forward without this logic.